src/FakeItEasy/Core/FakeManager.cs

```diff
@@ -103,6 +101,22 @@ public virtual void AddRuleLast(IFakeObjectCallRule rule)
         }
 
         /// <summary>
+        /// Adds a call rule to the fake object after the specified rule.
+        /// </summary>
+        /// <param name="previousRule">The rule after which to add a rule.</param>
+        /// <param name="newRule">The rule to add.</param>
+        public virtual void AddRuleAfter(IFakeObjectCallRule previousRule, IFakeObjectCallRule newRule)
+        {
+            var previousNode = this.AllUserRules.Nodes().FirstOrDefault(n => n.Value.Rule == previousRule);
+            if (previousNode == null)
+            {
+                throw new InvalidOperationException("The rule after which to add the new rule was not found in the list.");
+            }
+
+            this.AllUserRules.AddAfter(previousNode, new CallRuleMetadata { Rule = newRule });
+        }

```

Comment:
I wonder if I should make this internal... there's probably no harm in exposing it publicly, but if it's public, there should probably be an `AddRuleBefore` method as well. Let me know what you think.